jujugui/static/gui/src/app/jujulib/bakery.js

```diff
@@ -73,14 +71,28 @@ var module = module;
       this._client = httpClient;
       this.storage = storage;
       this._onSuccess = params.onSuccess || (() => {});
-      this._noRedirects = !!params.noRedirects;
-      if (params.nonInteractive) {
-        this._visitPage = this._nonInteractiveVisitPage;
-      } else {
-        this._visitPage = params.visitPage || (visitURL => {
-          window.open(visitURL, 'Login');
-        });
-      }
+      this._visitPage = params.visitPage || (visitURL => {
+        window.open(visitURL, 'Login');
+      });
+      this._dischargeDisabled = false;
```

Comment:
Because I didn't want users to be able to instantiate a bakery without discharge directly. As this is a really specific need that could even go away in the future, I think it's preferable to only have one way of creating such a bakery.